lib/online_order.rb

```diff
@@ -0,0 +1,103 @@
+require 'awesome_print'
+require 'pry'
+require 'csv'
+require_relative 'order'
+require_relative 'customer'
+
+
+module Grocery
+
+# Custom ArgumentError because it's fun, could also be done at the point of raising the ArgumentError using syntax ArgumentError.new("Custom Error Message")
+  class CartError < ArgumentError
+    def initialize(msg="Error: Items cannot be added to this order based on its current status")
+      super
+    end
+  end
+
+  class OnlineOrder < Order
+    attr_reader :customer, :status
+    @@all_online_orders = []
+    @@customers = []
+
+    def initialize(id, products, customer_id, status = :pending)
+      #Use the itialize attributes from the parent class Order
+      super(id, products)
+      @customer = customer_id
+      @status = status.to_sym
+    end # Initialize
+
+    def total
+      unless @products.length == 0
+        return super + 10
+      end
+    end
+
+    def add_product(product_name, product_price)
+      if @status == :paid || @status == :pending
+        super
+      else
+        begin
+          raise CartError
+        rescue
+        end
```

Comment:
These four lines do nothing. When you `raise` an error inside of a `begin` block, it will immediately be caught and drop you into the `rescue`. Since the `rescue` block is empty, this ends up being a no-op.

In general, you use `raise` to indicate an error happened and your code doesn't know what to do about it, whereas you use `begin`/`rescue` when you're calling some other method that you know might raise an error that you do know how to handle. Putting the two together is possible but very rare.